plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java

```diff
@@ -65,6 +66,59 @@ public synchronized void setModelState(String modelId, MLModelState state) {
         getExistingModelCache(modelId).setModelState(state);
     }
 
+    /**
+     * Set memory size estimation CPU/GPU
+     * @param modelId model id
+     * @param format model format like onnx
+     * @param size memory size
+     */
+    public synchronized void setMemSizeEstimation(String modelId, MLModelFormat format, Long size) {
+        Long memSize = getMemSizeEstimation(format, size);
+        log.debug("Updating memSizeEstimation of Model {}  to {}", modelId, memSize);
+        getExistingModelCache(modelId).setMemSizeEstimationCPU(memSize);
```

Comment:
I see we set same value for CPU and GPU, does that mean the CPU and GPU memory consumption is almost the same?

Reply:
I also have the same question. Github should have a `+1` icon :)

Reply:
Yes, from my experiments the cpu, gpu memory consumption is similar.